sdks/java/io/hdfs/src/main/java/org/apache/beam/sdk/io/hdfs/HadoopFileSystemOptions.java

```diff
@@ -39,11 +45,37 @@
   void setHdfsConfiguration(List<Configuration> value);
 
   /** A {@link DefaultValueFactory} which locates a Hadoop {@link Configuration}. */
-  class ConfigurationLocator implements DefaultValueFactory<Configuration> {
+  class ConfigurationLocator implements DefaultValueFactory<List<Configuration>> {
+    private static final Logger LOG = LoggerFactory.getLogger(ConfigurationLocator.class);
     @Override
-    public Configuration create(PipelineOptions options) {
-      // TODO: Find default configuration to use
-      return null;
+    public List<Configuration> create(PipelineOptions options) {
+      // Find default configuration when HADOOP_CONF_DIR or YARN_CONF_DIR is set.
+      Configuration conf = new Configuration(false);
+      List<String> hadoopEnvList = Lists.newArrayList("HADOOP_CONF_DIR", "YARN_CONF_DIR");
+      for (String env : hadoopEnvList) {
+        String hadoopConfPath = System.getenv(env);
```

Comment:
Add unit tests.

Note that `System.getenv` isn't mockable so the best bet is to make a method on `ConfigurationLocator` called:
```
@VisibleForTesting
Map<String, String> getEnvironment() {
  return System.getenv(key);
}
```

See [GcpOptions](https://github.com/apache/beam/blob/master/sdks/java/extensions/gcp-core/src/main/java/org/apache/beam/sdk/extensions/gcp/options/GcpOptions.java#L189) and [GcpOptionsTest](https://github.com/apache/beam/blob/master/sdks/java/extensions/gcp-core/src/test/java/org/apache/beam/sdk/extensions/gcp/options/GcpOptionsTest.java#L124) for an example of how this kind of interaction can be tested